### ISSUE TITLE
Building my own rotation, I get this error and can't create actions pretty much

### ISSUE THREAD

**Issue by hsramm** [KiNoYo/rotation-builder#7]
```
1x RotationBuilder\RotationBuilder-1.71.0.lua:1109: attempt to perform arithmetic on a nil value
RotationBuilder\RotationBuilder-1.71.0.lua:1109: in function `ROB_SpellValidate'
RotationBuilder\RotationBuilder-1.71.0.lua:1675: in function `ROB_Rotation_Edit_UpdateUI'
RotationBuilder\RotationBuilder-1.71.0.lua:921: in function `ROB_ActionListButton_OnClick'
[string "*:OnClick"]:1: in function <[string "*:OnClick"]:1>

Locals:
_spell = "<spell name>"
_spellingCheckPassed = false
_parsedSpellID = nil
_link = nil
(*temporary) = <function> defined =[C]:-1
(*temporary) = ""
(*temporary) = nil
(*temporary) = ""
(*temporary) = ":"
(*temporary) = nil
(*temporary) = "attempt to perform arithmetic on a nil value"
_InvSlots = <table> {
 WristSlot = 9
 BackSlot = 15
 Trinket0Slot = 13
 TabardSlot = 19
 FeetSlot = 8
 NeckSlot = 2
 Finger1Slot = 12
 ChestSlot = 5
 Trinket1Slot = 14
 AmmoSlot = 0
 MainHandSlot = 16
 RangedSlot = 18
 SecondaryHandSlot = 17
 ShirtSlot = 4
 Finger0Slot = 11
 HandsSlot = 10
 LegsSlot = 7
 WaistSlot = 6
 ShoulderSlot = 3
 HeadSlot = 1
}
ROB_EditingRotationTable = <table> {
 bindindex = 0
 SortedActions = <table> {
 }
 ActionList = <table> {
 }
 specID = 1
 isMultiTarget = false
}
ROB_CurrentActionName = "Keg Smash"
```

**Comment by vdubus** [comment]
Hello, I would like to have a bit more information on how to reproduce this problem.
Could you do the following actions to help us understand this?

1. Disable all other addon to make sure that's not some kind of conflict.
2. Clean-up all `RotationBuilder.lua` file in theses directories: `<WoW installation directory>\WTF\Account\<Account name>\Les Clairvoyants\<Each Characters>\SavedVariables`.
3. Uninstall and reinstall Rotation Builder. Check the `<WoW Installation Directory>\Interface\AddOns` directory to make sure that Rotation Builder was correctly uninstalled.
4. Try again and describe us how you are producing this bug. Add some screenshots of the rotation you are trying to make to help us understand how the bug occur.

With all these steps, we want to make sure that we will be able to reproduce this bug correctly so we can fix it.

**Comment by hsramm** [comment]
Trying to add an Action in my own custom rotation that I am building. Everytime I add an action, I get that error and if it's a brand new rotation and there is nothing there...it won't show an options panel for the action at all.

**Comment by jdubus** [comment]
It seems to me from the error message to be a case of invalid spell. As the spell doesn't exist it isn't valid but there are checks for that, it should normally just say that the spell isn't valid and not bother with it until it is valid. So maybe Blizzard changed something (again) in the way some APÏ functions works in one of the latest patches. It wouldn't be the first time and certainly won't be the last.

**Comment by vdubus** [comment]
A quick test gave me the same kind of exception.
I will check that.

```
1x RotationBuilder\RotationBuilder-1.72.0.lua:1109: attempt to perform arithmetic on a nil value
RotationBuilder\RotationBuilder-1.72.0.lua:1109: in function `ROB_SpellValidate'
RotationBuilder\RotationBuilder-1.72.0.lua:1675: in function `ROB_Rotation_Edit_UpdateUI'
RotationBuilder\RotationBuilder-1.72.0.lua:1057: in function `OKCallbackFunction'
RotationBuilder\RotationBuilder-1.72.0.lua:1213: in function <RotationBuilder\RotationBuilder.lua:1211>

Locals:
_spell = "<spell name>"
_spellingCheckPassed = false
_parsedSpellID = nil
_link = nil
(*temporary) = <function> defined =[C]:-1
(*temporary) = ""
(*temporary) = nil
(*temporary) = ""
(*temporary) = ":"
(*temporary) = nil
(*temporary) = "attempt to perform arithmetic on a nil value"
_InvSlots = <table> {
 WristSlot = 9
 BackSlot = 15
 Trinket0Slot = 13
 TabardSlot = 19
 FeetSlot = 8
 NeckSlot = 2
 Finger1Slot = 12
 ChestSlot = 5
 Trinket1Slot = 14
 AmmoSlot = 0
 MainHandSlot = 16
 RangedSlot = 18
 SecondaryHandSlot = 17
 ShirtSlot = 4
 Finger0Slot = 11
 HandsSlot = 10
 LegsSlot = 7
 WaistSlot = 6
 ShoulderSlot = 3
 HeadSlot = 1
}
ROB_EditingRotationTable = <table> {
 ActionList = <table> {
 }
 SortedActions = <table> {
 }
 bindindex = 0
}
ROB_CurrentActionName = "test"

```

**Comment by vdubus** [comment]
The bug was caused by an update made by Blizzard in their API.
This change broke our usage of the function `GetSpellLink`.
Check commit 5a2d62e8d070e0e9114bed762c1571413efdb852.

Fixed in release 1.72.0. 
Currently waiting Curse build for download.
